internal/wasm/jit/engine.go

```diff
@@ -38,10 +37,8 @@ type (
 		parentEngine           *engine
 		importedFunctionCounts int
 
-		// Closed indicates whether Close was called for this moduleEngine, and used to prevent
-		// double-free of compiledFunctions.
-		// Note: intentionally use uint32 instead of bool for atomic operation.
-		closed uint32
+		// closed is the field updated atomically that guards Close
+		closed uint64
```

Comment:
I'm secretly hoping we can load64 this in assembly to quit wasm early when non-zero.